src/system/SystemPacketBuffer.cpp

```diff
@@ -605,14 +605,30 @@ PacketBufferHandle PacketBufferHandle::PopHead()
     return PacketBufferHandle(head);
 }
 
-PacketBufferHandle PacketBufferHandle::CloneData(uint16_t aAdditionalSize, uint16_t aReservedSize)
+PacketBufferHandle PacketBufferHandle::CloneData()
 {
-    if (!mBuffer->Next().IsNull())
+    PacketBufferHandle cloneHead;
+
+    for (PacketBuffer * original = mBuffer; original != nullptr; original = static_cast<PacketBuffer *>(original->next))
     {
-        // We do not clone an entire chain.
-        return PacketBufferHandle();
+        uint16_t originalDataSize     = original->MaxDataLength();
+        uint16_t originalReservedSize = original->ReservedSize();
+        PacketBufferHandle clone      = PacketBufferHandle::New(originalDataSize, originalReservedSize);
```

Comment:
Can this construction fail?

Reply:
It can, but then `clone.IsNull()` will test true and we will take our early return.

Reply:
This was added to address my comment.

But yes, with the fix, looks good.